src/verifier/client.rs

```diff
@@ -43,17 +43,18 @@ impl DIDClient {
     pub async fn new(
         vm: String,
         signer: Arc<dyn RequestSigner + Send + Sync>,
-        resolver: &dyn DIDResolver,
+        resolver: impl JWKResolver,
     ) -> Result<Self> {
         let (id, _f) = vm.rsplit_once('#').context(format!(
             "expected a DID verification method, received '{vm}'"
         ))?;
 
-        let key = ssi::did_resolve::resolve_key(&vm, resolver)
+        let key = resolver
+            .fetch_public_jwk(Some(&vm))
             .await
             .context("unable to resolve key from verification method")?;
 
-        if &key != signer.jwk() {
+        if &*key != signer.jwk() {
```

Comment:
Used the `JWKResolver` since this check is performed, but not sure this is the correct key resolution process.

Reply:
As above, I think this is correct, you just need to pass in the correct DID resolver.

Reply:
Perfect, that was the issue! Thanks for the tip!